src/curl.c

```diff
@@ -62,6 +62,33 @@ static void swupd_curl_strerror(CURLcode curl_ret)
 	fprintf(stderr, "Curl error: (%d) %s\n", curl_ret, curl_easy_strerror(curl_ret));
 }
 
+/* 
+ * Cannot avoid a TOCTOU here with the current curl API.  Only using the curl
+ * API, a curl_easy_setopt does not detect if the client SSL certificate is
+ * present on the filesystem.  The only happens during curl_easy_perform.
+ * The emphasis is rather on how using a SSL client certificate is an opt-in
+ * function rather than an opt-out function.
+ */
+CURLcode swupd_curl_set_optional_client_cert(CURL* curl) {
+	CURLcode curl_ret = CURLE_OK;
+	char* client_cert_path;
+
+	client_cert_path = mk_full_filename(path_prefix, SSL_CLIENT_CERT);
```

Comment:
You need to free client_cert_path after using it.

Reply:
yes, I missed this.